sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/SelectByteBuddyHelpers.java

```diff
@@ -177,7 +178,9 @@ static RowSelector createRowSelector(SchemaAndDescriptor schemaAndDescriptor) {
       return builder
           .visit(new AsmVisitorWrapper.ForDeclaredMethods().writerFlags(ClassWriter.COMPUTE_FRAMES))
           .make()
-          .load(Row.class.getClassLoader(), ClassLoadingStrategy.Default.INJECTION)
+          .load(
+              ReflectHelpers.findClassLoader(schemaAndDescriptor.getClass().getClassLoader()),
+              getClassLoadingStrategy(schemaAndDescriptor.getClass()))
```

Comment:
I'm not sure if we want the change to use `schemaAndDescriptor.getClass()` instead of `Row.class`, or if this should also be reverted

Reply:
I'm going to go ahead and revert this change, tentatively resolving